vcstool/commands/compare.py

```diff
@@ -1,80 +1,595 @@
 
-    # Get the repos from the repo file
+    def __init__(self, path: str) -> None:
+        self._path = path
+
+    def is_significant(self) -> bool:
+        return True
+
+    def legend_flags(self) -> Legend.Flags:
+        return Legend.Flags.REPO_STATUS | Legend.Flags.MISSING_REPO
+
+    def get_color_repo_status(self) -> str:
+        return ansi("redf") + Legend.MISSING
+
+    def get_color_repository(self) -> str:
+        return ansi("redf") + self._path
+
+    def get_color_branch(self) -> str:
+        return ansi("redf") + "ABSENT FROM FILESYSTEM"
+
+    def get_color_track(self) -> str:
+        return ""
+
+    def get_color_vcs_tracking_flags(self) -> str:
+        return ""
+
+    def get_color_tag(self) -> str:
+        return ""
+
+    def get_color_hash(self) -> str:
+        return ""
+
+
+class CompareOutputEntry(ICompareTableEntry):
+    """Entry for a repo which is discovered by the CompareCommand, but may be missing from the
+    manifest (i.e. `manifest_version is None`)"""
+
+    def __init__(
+        self,
+        path: str,
+        compare_output: CompareOutput,
+        manifest_version: Optional[str] = None,
+    ) -> None:
+        self._path = path
+        self._compare_output = compare_output
+        self._manifest_version = manifest_version
+        self._is_current_with_manifest = self._manifest_version in [
+            self._compare_output.local_branch,
+            self._compare_output.hash,
+        ]
+
+        self._vcs_tracking_flags = self._get_vcs_tracking_flags()
+        self._repo_status = self._get_repo_status()
+        self._vcs_tracking_status = self._get_tracking_status()
+
+    def is_significant(self) -> bool:
+        return any(
+            [
+                self._is_dirty(),
+                self._vcs_tracking_status != VcsTrackingStatus.EQUAL,
+                # TODO(amusco): is empty git repo?
+            ]
+        )
+
+    def legend_flags(self) -> Legend.Flags:
+        flags = Legend.Flags(0)
+        if (
+            self._vcs_tracking_status != VcsTrackingStatus.EQUAL
+            or self._vcs_tracking_flags.strip() != ""
+        ):
+            flags |= Legend.Flags.VCS_TRACKING_STATUS
+        if self._repo_status != RepoStatus.NOMINAL:
+            flags |= Legend.Flags.REPO_STATUS
+        return flags
+
+    def get_color_repo_status(self) -> str:
+        return {
+            RepoStatus.NOMINAL: ansi("brightblackf") + Legend.SUBMODULE,
+            RepoStatus.UNTRACKED: ansi("brightyellowf") + Legend.REPO_NOT_TRACKED,
+        }[self._repo_status]
+
+    def get_color_repository(self) -> str:
+        foreground = ansi("brightcyanf") if self.is_significant() else ansi("whitef")
+        return foreground + self._path
+
+    def get_color_branch(self) -> str:
+        foreground = ansi("white")
+        if not self._is_valid_branch_name(self._compare_output.local_branch):
```

Comment:
I think we can leave this check if we move this to a fork within Outrider's repository, otherwise we might want to remove as branch naming is specific to our company.

Reply:
Yeah definitely felt weird about adding this to the public repo. Perhaps read an optional config file specifying valid branch names?

Reply:
It is probably not worth the time to implement a optional config file to specify valid branch names, if we were to poll developers on how many people recognized when a branch name was invalid I would say less than a handful would know. I say lets remove functionality for now and we can always add something more customizable later. Does that work?